### PR TITLE
streamingccl: add telemetry counter for started replication streams

### LINKED ISSUE #120232: Add telemetry feature counter for `CREATE VIRTUAL CLUSTER  FROM REPLICATION`
The `CREATE VIRTUAL CLUSTER FROM REPLICATION` statement should increment a feature counter.

Jira issue: CRDB-36561

### PR DESCRIPTION
**Pull request by stevendanna**
Fixes #120232

Release note: None

### CODE REVIEW

**Comment by stevendanna** [comment]
@alicia-l2 Let me know if there are other counters you may want. This just adds a counter for started streams.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go [hunk 1]
```diff
@@ -562,6 +566,11 @@ func TestCutoverCheckpointing(t *testing.T) {
 
 	jobutils.WaitForJobToRun(c.T, c.SrcSysSQL, jobspb.JobID(producerJobIDInt))
 	jobutils.WaitForJobToRun(c.T, c.DestSysSQL, replicationJobID)
+
+	// Assert that our counters have been incremented.
+	counts := telemetry.GetFeatureCounts(telemetry.Raw, telemetry.ResetCounts)
+	require.GreaterOrEqual(t, counts["physical_replication.started"], int32(1))
```
Comment (msbutler) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go:572]: why isn't this `require.Equal()`?

**Comment by alicia-l2** [comment]
What about adding counters for cutover and flashback as well?

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go (hunk 1 above)
Comment (stevendanna) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go:572]: "something something txn retries"

Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go:572]: wait, but even if we increment when `err == nil`?? Thats odd. I wish we could test that this thing only gets incremented 1 time, most of the time.

**Comment by stevendanna** [comment]
@alicia-l2 @msbutler This is ready for a look.

**Review by msbutler — APPROVED** [review]

**Review by alicia-l2 — APPROVED** [review]

**Comment by rickystewart** [comment]
Pre-emption.

bors r=stevendanna

**Pull request merged**